### PR TITLE
openshift/from-source now works with latest linch-pin

### PR DESCRIPTION
**Pull request by herlo**
openshift/from-source now works with linch-pin, includes fixes to playbooks, jjb, and topology.
Also, it now emails upon failure and fixed scenarios (thanks @arilivigni)

### CODE REVIEW

File: jjb/openshift/from-source/ofs-macros.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@
     site: duffy
     jslave_name: paas-sig-ci-slave01
     jslave_username: root
+    lpsha1: 77e5613227df670efc5e877134fce67432e5e97d
```
Comment (arilivigni) [inline jjb/openshift/from-source/ofs-macros.yaml]: Will this be removed after added to main lp?

File: jjb/openshift/openshift-defaults.yml [hunk 2]
```diff
@@ -9,6 +15,13 @@
             properties-content: |
                 ANSIBLE_SSH_CONTROL_PATH=%(directory)s/%%h-%%r
                 ANSIBLE_HOST_KEY_CHECKING=False
+    properties:
+        - build-discarder:
+            num-to-keep: 50
+            artifact-num-to-keep: 15
+        - inject:
+            properties-content: |
+              lp_sha1=${lpsha1}
```
Comment (arilivigni) [inline jjb/openshift/openshift-defaults.yml]: Shouldn't this be lp_sha1={lpsha1}

File: jjb/openshift/openshift-defaults.yml [hunk 3]
```diff
@@ -1,5 +1,11 @@
 - defaults:
     name: openshift-defaults
+    description: |
+        Maintainer: <a href="mailto:herlo@redhat.com>Clint Savage</a>
```
Comment (arilivigni) [inline jjb/openshift/openshift-defaults.yml]: you could also add me as a second-maintainer

File: jjb/openshift/openshift-defaults.yml (hunk 2 above)
Comment (herlo) [inline jjb/openshift/openshift-defaults.yml]: I guess both ways work. But consistency is key. :+1:

File: jjb/openshift/from-source/ofs-macros.yaml (hunk 1 above)
Comment (arilivigni) [inline jjb/openshift/from-source/ofs-macros.yaml]: agreed

File: jjb/openshift/openshift-defaults.yml (hunk 3 above)
Comment (arilivigni) [inline jjb/openshift/openshift-defaults.yml]: this one?

Comment (herlo) [inline jjb/openshift/openshift-defaults.yml]: You're there, just look lower in the file, one line down. :)

**Pull request merged**